src/python_testing/matter_testing_infrastructure/chip/testing/metadata.py

```diff
@@ -119,6 +127,17 @@ def extract_runs_arg_lines(py_script_path: str) -> Dict[str, Dict[str, str]]:
             elif args_match:
                 runs_arg_lines[args_match.group("run_id")][args_match.group("arg_name")] = args_match.group("arg_val")
 
+    if not runs_arg_lines:
+        try:
+            runs = yaml.safe_load("\n".join(ci_args_section_lines))
+            for run, args in runs.get("test-runner-runs", {}).items():
+                runs_arg_lines[run] = {}
+                runs_arg_lines[run]['run'] = run
+                runs_arg_lines[run]['py_script_path'] = py_script_path
+                runs_arg_lines[run].update(args)
+        except yaml.YAMLError as e:
+            print(f"ERROR: {py_script_path}: Failed to parse CI arguments YAML: {e}", file=sys.stderr)
```

Comment:
please use logging.error or similar. Printing out to stderr is frustrating to use because you have to do extra work to grep.

maybe printing the stack trace would also help here (via the traceback module)

Reply:
Replaced all prints to logging.

> maybe printing the stack trace would also help here (via the traceback module)

Printing stack trace will not add any meaningful info in that case. I've modified error reporting a little bit, so right now the error will include proper line and column within the parsed test script, e.g:

```
$ ./scripts/tests/run_python_test.py --script src/python_testing/TC_ECOINFO_2_1.py --load-from-env=test-env.yaml
ERROR:root:Failed to parse CI arguments YAML: mapping values are not allowed here
  in "src/python_testing/TC_ECOINFO_2_1.py", line 24, column 10
Traceback (most recent call last):
  File ".../connectedhomeip/./scripts/tests/run_python_test.py", line 238, in <module>
    main(auto_envvar_prefix='CHIP')
  File ".../connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/a.bokowy/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File ".../connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File ".../connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File ".../connectedhomeip/./scripts/tests/run_python_test.py", line 118, in main
    raise Exception(
Exception: No valid runs were found. Make sure you add runs to your file, see https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md document for reference/example.
```